src/middleware/error-handler/error-handler.ts

```diff
@@ -0,0 +1,11 @@
+import { HttpError } from '../../errors/http-error'
+import { type Request, type Response, type NextFunction } from 'express'
+import { ApiErrors } from '../../enums/API-errors.enum'
+
+const errorHandler = (err: Error, req: Request, res: Response, next: NextFunction): void => {
+    const serverError = new HttpError(ApiErrors.SERVER_ERROR, 500)
+    res.status(500).json({ error: serverError })
```

Comment:
error handler middleware have to respond with appropriate error codes(400, 404, 422) not only 500.